client-mr/src/main/java/org/apache/hadoop/mapreduce/v2/app/RssMRAppMaster.java

```diff
@@ -199,6 +200,24 @@ public Thread newThread(Runnable r) {
       extraConf.set(RssMRConfig.RSS_REMOTE_STORAGE_PATH, remoteStorage.getPath());
       extraConf.set(RssMRConfig.RSS_REMOTE_STORAGE_CONF, remoteStorage.getConfString());
 
+      // When containers have disk with very limited space, reduce is allowed to spill data to hdfs
+      if (extraConf.getBoolean(RssMRConfig.RSS_REDUCE_REMOTE_SPILL_ENABLED,
+        RssMRConfig.RSS_REDUCE_REMOTE_SPILL_ENABLED_DEFAULT)) {
+
+        if (storageType != StorageType.MEMORY_LOCALFILE_HDFS.name()) {
```

Comment:
Could we check check remote storage path instead?

Reply:
ok